src/GraphQL.Harness/GraphQLMiddleware.cs

```diff
@@ -51,7 +53,7 @@ private async Task ExecuteAsync(HttpContext context, ISchema schema)
         {
             var start = DateTime.UtcNow;
 
-            var request = await context.Request.Body.FromJsonAsync<GraphQLRequest>(context.RequestAborted);
+            var request = await _reader.ReadAsync<GraphQLRequest>(context.Request.Body, context.RequestAborted);
```

Comment:
Here you can see in the basic sample provided where the middleware code was tied directly to the `GraphQL.SystemTextJson` extension method using an unconfigurable JSON serializer.

Now the code uses the registered reader, which can be configured elsewhere in the system.

(Note: the old code works also.)

Reply:
> where the middleware code was tied directly to the GraphQL.SystemTextJson extension method using an unconfigurable JSON deserializer.

It's just a sample project.